src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -546,7 +546,8 @@ trait Applications extends Compatibility { self: Typer with Dynamic =>
       val fun1 = typedExpr(tree.fun, originalProto)
 
       // Warning: The following lines are dirty and fragile. We record that auto-tupling was demanded as
-      // a side effect in adapt. If it was, we assume the tupled proto-type in the rest of the application.
+      // a side effect in adapt. If it was, we assume the tupled proto-type in the rest of the application,
+      // until, possibly, we have to fall back to insert an implicit on thq qualifier.
```

Comment:
So why does the code do `tryWithImplicitOnQualifier(fun1, originalProto)` and then fallbacks to `tryWithImplicitOnQualifier(fun1, proto)`, isn't `originalProto` the non-tupled one?